### DISCUSSION TITLE
static adapter and fetching data

### DISCUSSION THREAD

**Discussion by vorcigernix** [sveltejs/kit#7334] [Q&A] [answered]
I moved the static adapter in order to prerender all the pages and it works beautifully. The problem is that the data in external json is fetched at build time and they are not updated unless I rebuild the site. Is there a way to keep all my pages static except specific routes that will be hydrated? I understand that this requires javascript and I am fine with that on those specific pages. 
I think that what I need is some specific configuration enabling SPA mode, but I can't figure out how to do that (all my attempts ended with all pages in SPA mode).

**Comment by teemingc** [comment, marked as answer]
Have you tried fetching the data in an `onMount()` function or using an `{#await}` block?
(Not sure if this works as I haven't done this before)

**Comment by vorcigernix** [comment]
Yeah, that is actually the solution I just deployed, now I need to test if
it actually works. I'll share the link to the repo if I succeed.

**Comment by vorcigernix** [comment]
So yes, it is working, rewriting data object onmount seems to be a good option. I was also considering service workers, but this is simple and working. Example code is in https://github.com/vorcigernix/ethbrno_workshop/tree/swr-markdown/src/routes (+page files).

**Comment by dummdidumm** [comment]
An alternative to `onMount` is to add `export const prerender = false` in `+page.js` only to the pages that you don't want prerendered. That along with setting the fallback option of adapter-static should do it.
